packages/lesswrong/components/common/HashLink.tsx

```diff
@@ -50,7 +58,7 @@ function hashLinkScroll() {
   }, 0);
 }
 
-export function genericHashLink(props, As) {
+export function genericHashLink(props: HashLinkProps) {
```

Comment:
Having a critical component like `Link` go unnecessarily untyped was driving me a bit batty, so... fixed.

Reply:
Turns out we weren't using `NavLink` in a single place in the codebase!  So got rid of it (along with all the additional complexity it added here).